src/services/transactionService.ts

```diff
@@ -249,21 +258,30 @@ interface IBuildSorobanTransferOperation {
   token: SdkToken;
   transactionBuilder: TransactionBuilder;
   network: NETWORKS;
+  memo?: string; // Optional memo for creating muxed address
+  contractSupportsMuxed?: boolean; // Whether contract supports muxed addresses
 }
 
 /**
  * Builds a Soroban token transfer operation for sending to contract addresses
+ * Supports muxed addresses (M... format) for CAP-0067 memo support
+ *
+ * @param params Transfer operation parameters
+ * @returns Final destination address (may be muxed if memo was provided and contract supports it)
+ * @note transactionBuilder is mutated in place, so it doesn't need to be returned
  */
 export const buildSorobanTransferOperation = (
   params: IBuildSorobanTransferOperation,
-): TransactionBuilder => {
```

Comment:
looks like buildSorobanTransferOperation is only being used locally here, so we could remove the `export` to make it easier to review/debug